src/python/grpcio/tests/protoc_plugin/protos/service/test_service.proto

```diff
@@ -62,3 +62,34 @@ service TestService {
   rpc HalfDuplexCall(stream StreamingOutputCallRequest)
       returns (stream StreamingOutputCallResponse);
 }
+
+service MassagedNames1 {
+  // -> 'anametest'
+  rpc Anametest(SimpleRequest) returns (SimpleResponse);
+}
+
+service MassagedNames2 {
+  // -> 'nametest'
+  rpc NAMETEST(SimpleRequest) returns (SimpleResponse);
+}
+
+service MassagedNames3 {
+  // -> 'name_test1'
+  rpc NAMETest1(SimpleRequest) returns (SimpleResponse);
+}
+
+service MassagedNames4 {
+  // -> 'name_tes_t1' (we *only* use capital letters to distinguish word
```

Comment:
I *believe* that's currently followed by virtue of non-uppercase letters never being used to create a word boundary.